jerry-core/ecma/operations/ecma-exceptions.c

```diff
@@ -140,23 +144,44 @@ ecma_new_standard_error (ecma_standard_error_t error_type) /**< native error typ
 
   ((ecma_extended_object_t *) new_error_obj_p)->u.class_prop.class_id = LIT_MAGIC_STRING_ERROR_UL;
 
-#if ENABLED (JERRY_LINE_INFO)
-  /* The "stack" identifier is not a magic string. */
-  const char * const stack_id_p = "stack";
+  if (message_string_p != NULL)
+  {
+    ecma_property_value_t *prop_value_p;
+    prop_value_p = ecma_create_named_data_property (new_error_obj_p,
+                                                    ecma_get_magic_string (LIT_MAGIC_STRING_MESSAGE),
+                                                    ECMA_PROPERTY_CONFIGURABLE_WRITABLE,
+                                                    NULL);
+
+    ecma_ref_ecma_string (message_string_p);
+    prop_value_p->value = ecma_make_string_value (message_string_p);
+  }
 
-  ecma_string_t *stack_str_p = ecma_new_ecma_string_from_utf8 ((const lit_utf8_byte_t *) stack_id_p, 5);
+  /* Avoid calling the decorator function recursively. */
+  if (JERRY_CONTEXT (error_object_update_callback_p) != NULL
+      && !(JERRY_CONTEXT (status_flags) & ECMA_STATUS_ERROR_UPDATE))
+  {
+    JERRY_CONTEXT (status_flags) |= ECMA_STATUS_ERROR_UPDATE;
+    JERRY_CONTEXT (error_object_update_callback_p) (ecma_make_object_value (new_error_obj_p));
```

Comment:
Are you sure that whenever the decorator callback is active, no `stack` property need to be assigned to the error object?

Reply:
It is up to the application which is using JerryScript. It can create such property in any way it wants.